src/components/scrollable-list/test-utils.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import ScrollableListItem from './scrollable-list-item.component';
+import 'jest';
+
+const makeArray = n => [...Array(n).keys()];
+
+const listItemReducer = ({ nonSelectables = [], customSelectables = [] }) => {
+  return (acc, item) => {
+    if (nonSelectables.includes(item)) {
+      return [...acc, <div />];
+    } else if (customSelectables.includes(item)) {
+      return [...acc, <div isSelectable></div>]
+    }
+
+    return [...acc, <ScrollableListItem />];
+  };
+};
+
+const renderListItems = (opts = {}) => {
+  return makeArray(opts.num).reduce(listItemReducer(opts), []);
+};
+
+
+const keyPress = (code) => { 
+  const ev = new KeyboardEvent('keydown', { which: code });
+  document.dispatchEvent(ev);
+};
+
+const keyMap = {
+  UpArrow: '38',
+  DownArrow: '40',
+  RightArrow: '39',
+  LeftArrow: '37',
+  Enter: '13'
+};
+
+const repeat = action => (n = 1) => makeArray(n).forEach(() => action());
+
+const keyboard = Object.keys(keyMap).reduce((acc, key) => {
+  acc[`press${key}`] = () => repeat(keyPress(keyMap[key]));
+  return acc;
+}, {});
+
+const listFrom = wrapper => wrapper.find('ul');
+const childrenFrom = node => node.children();
+
+const hoverList = wrapper => (item) => {
+  childrenFrom(listFrom(wrapper)).at(item).simulate('mouseover');
+};
+
+const simulateEvent = eventName => wrapper => wrapper.simulate(eventName);
+const click = simulateEvent('click');
+
+const traverseListObject = length => () => keyboard.pressDownArrow(length);
+const selectedItemOf = wrapper => wrapper.state().selectedItem;
+const isUnique = (val, index, self) => self.indexOf(val) === index;
+const isSelectableGiven = nonSelectables => i => !nonSelectables.includes(i);
+
+const selectedItemReducer = method => wrapper => (acc, i) => {
+  method(wrapper)(i);
+  return [...acc, selectedItemOf(wrapper)];
+};
+
+const arraysEqual = (arr1, arr2) => arr1.sort().join(',') === arr2.sort().join(',');
+
+const assertCorrectTraversal = method => expect => ({ num, nonSelectables = [] }) => (wrapper) => {
+  const array = makeArray(num);
+  const validIndexes = array.filter(isSelectableGiven(nonSelectables));
+
+  const indexesThatWereSelected = array
+    .reduce(selectedItemReducer(method)(wrapper), [])
+    .filter(isUnique);
+  expect(arraysEqual(validIndexes, indexesThatWereSelected)).toBeTruthy();
+};
+
+const assertKeyboardTraversal = assertCorrectTraversal(() => keyboard.pressDownArrow)(expect);
+const assertHoverTraversal = assertCorrectTraversal(wrapper => hoverList(wrapper))(expect);
+
+export {
+  hoverList,
+  selectedItemOf,
+  childrenFrom,
+  renderListItems,
+  keyboard,
+  assertKeyboardTraversal,
+  assertHoverTraversal,
+  traverseListObject,
+  listFrom,
+  click
+};
```

Comment:
Agreed that helpers like this should be shared across the library. The intention with this was to offer some suggestions as to how we can clean up our approach to writing these unit tests, rather than a finished solution. 

I argue that we should try to improve our approach in making unit tests more thorough and more readable. Declaring more at the top of test blocks and keeping individual `it` blocks as thin as possible, avoiding wherever possible testing instance methods on the components directly, etc. 

Also, I'm finding a lot of instances where we're having to alter the code in our components to work around the limitations of the Jest API and JSDom.